src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorConfigurationService.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorConfigurationService
+    {
+        private readonly ILanguageServer _server;
+        private readonly ILogger _logger;
+
+        public RazorConfigurationService(ILanguageServer languageServer, ILoggerFactory loggerFactory)
+        {
+            if (languageServer is null)
+            {
+                throw new ArgumentNullException(nameof(languageServer));
+            }
+
+            if (loggerFactory is null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _server = languageServer;
+            _logger = loggerFactory.CreateLogger<RazorConfigurationService>();
+        }
+
+        public async Task<RazorLSPOptions> GetLatestOptions()
+        {
+            try
+            {
+                var request = new ConfigurationParams()
+                {
+                    Items = new[]
+                    {
+                        new ConfigurationItem()
+                        {
+                            Section = "razor"
+                        },
+                    }
+                };
+
+                var result = await _server.Client.SendRequest<ConfigurationParams, object[]>("workspace/configuration", request);
+                if (result == null || result.Length < 1)
+                {
+                    return null;
+                }
+
+                var jsonString = result[0].ToString();
```

Comment:
Are we guaranteed to always have **only** 1? Also does the spec state that it's absolutely json?

Reply:
From [the LSP spec](https://microsoft.github.io/language-server-protocol/specification#workspace_configuration)
`The request can fetch several configuration settings in one roundtrip. The order of the returned configuration settings correspond to the order of the passed ConfigurationItems (e.g. the first item in the response is the result for the first configuration item in the params).`

So if I'm reading that correctly the `result[0]` assumption is safe according to the spec.

Reply:
> Are we guaranteed to always have only 1?

What @ryanbrandenburg said.

> Also does the spec state that it's absolutely json?

No it doesn't explicitly say that. The return type is just `any[]`. But the spec says the following,

_The configuration section ask for is defined by the server and doesn’t necessarily need to correspond to the configuration store used be the client. So a server might ask for a configuration cpp.formatterOptions but the client stores the configuration in a XML store layout differently. It is up to the client to do the necessary conversion._

I think it is safe to assume Json in our case because the OmniSharp LSP implementation uses json.

Reply:
kk